cmd/argo/lint/lint.go

```diff
@@ -78,12 +86,45 @@ func GetFormatter(fmtr string) (Formatter, error) {
 	return f, nil
 }
 
+// RunLint lints the specified kinds in the specified files and prints the results to os.Stdout.
+// If linting fails it will exit with status code 1.
+func RunLint(ctx context.Context, client apiclient.Client, files, kinds []string, defaultNs, output string, strict bool) {
+	fmtr, err := GetFormatter(output)
+	errors.CheckError(err)
+
+	clients, err := getLintClients(client, kinds)
+	errors.CheckError(err)
+
+	res, err := Lint(ctx, &LintOptions{
+		ServiceClients:   clients,
+		Files:            files,
+		Strict:           strict,
+		DefaultNamespace: defaultNs,
+		Formatter:        fmtr,
+		Printer:          os.Stdout,
+	})
+	errors.CheckError(err)
+
+	if !res.Success {
+		os.Exit(1)
+	}
+}
+
 // Lint reads all files, returns linting errors of all of the enitities of the specified kinds.
 // Entities of other kinds are ignored.
 func Lint(ctx context.Context, opts *LintOptions) (*LintResults, error) {
+	var fmtr Formatter = defaultFormatter
+	var w io.Writer = ioutil.Discard
+	if opts.Formatter != nil {
+		fmtr = opts.Formatter
```

Comment:
I’d probably make these both mandatory, but provide null defaults.

Reply:
that's what I have now, no?